include/ignition/rendering/BoundingBoxCamera.hh

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef IGNITION_RENDERING_BOUNDINGBOXCAMERA_HH_
+#define IGNITION_RENDERING_BOUNDINGBOXCAMERA_HH_
+
+#include <cstdint>
+#include <vector>
+
+#include <ignition/common/Event.hh>
+#include <ignition/math/Color.hh>
+#include <ignition/math/Vector3.hh>
+
+#include "ignition/rendering/Camera.hh"
+
+namespace ignition
+{
+  namespace rendering
+  {
+    /// \brief BoundingBox types for Visible / Full 2D Boxes / 3D Boxes
+    enum class BoundingBoxType
+    {
+      /// 2D box that shows the full box of occluded objects
+      BBT_FULLBOX2D = 0,
+
+      /// 2D box that shows the visible part of the
+      /// occluded object
+      BBT_VISIBLEBOX2D = 1,
+
+      /// 3D oriented box
+      BBT_BOX3D = 2
+    };
+
+    /// \brief 2D or 3D Bounding box. It stores the
+    /// position / orientation / size info of the box and its label
+    struct BoundingBox
```

Comment:
I don't remember why we went with a custom struct, but I'll just bring up again that we could use `ignition::math::Box` (it doesn't have a `Vertices` function yet though :confused:

Reply:
What if we had a struct like this:
```c++
struct BoundingBox
{
  /// \brief The bounding box
  ignition::math::Boxd box;

  /// \brief The label of the object represented by the bounding box
  uint32_t label;

  /// \brief Get the vertices of the box
  std::vector<math::Vector3d> Vertices() const
};
```

That way, we avoid some of the duplication but still add things like `label` and `Vertices` that `ignition::math::Box` doesn't currently have.

Reply:
I like that

Reply:
We can decide this later now that the `BoundingBox` class is PIMPLized.